contracts/sol6/Dao/mock/KyberFeeHandlerWrapper.sol

```diff
@@ -0,0 +1,228 @@
+            startTokenIndex > endTokenIndex ||
+            startKyberFeeHandlerIndex > endKyberFeeHandlerIndex ||
+            supportedTokens.length == 0
+        ) {
+            // no need to do anything
+            return amounts;
+        }
+
+        uint256 endTokenId = (endTokenIndex >= supportedTokens.length) ?
+            supportedTokens.length : endTokenIndex;
+
+        for (uint256 i = startTokenIndex; i < endTokenId; i++) {
+            KyberFeeHandlerData[] memory kyberFeeHandlerArray = kyberFeeHandlersPerToken[supportedTokens[i]];
+            uint256 endKyberFeeHandlerId = (endKyberFeeHandlerIndex >= kyberFeeHandlerArray.length) ?
+                kyberFeeHandlerArray.length : endKyberFeeHandlerIndex;
+            amounts = new uint256[](endKyberFeeHandlerId - startKyberFeeHandlerIndex + 1);
+            
+            for (uint256 j = startKyberFeeHandlerIndex; j < endKyberFeeHandlerId; j++) {
+                ExtendedIKyberFeeHandler feeHandler = kyberFeeHandlerArray[j].kyberFeeHandler;
+                if (feeHandler.rebatePerWallet(rebateWallet) > 1) {
+                    amounts[j] = feeHandler.claimReserveRebate(rebateWallet);
+                }
+            }
+        }
+    }
+
+    /// @dev claim accumulated fee per platform wallet. Called by any address
+    /// @param platformWallet the wallet to claim fee for. Total accumulated fee sent to this wallet
+    /// @param startTokenIndex index of supportedTokens to start iterating from (inclusive)
+    /// @param endTokenIndex index of supportedTokens to start iterating from (exclusive)
+    /// @param startKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (inclusive)
+    /// @param endKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (exclusive)
+    /// @return amounts platform fee wei / twei amount claimed from each feeHandler
+    function claimPlatformFee(
+        address platformWallet,
+        uint256 startTokenIndex,
+        uint256 endTokenIndex,
+        uint256 startKyberFeeHandlerIndex,
+        uint256 endKyberFeeHandlerIndex
+    ) external returns (uint256[] memory amounts)
+    {
+        if (
+            startTokenIndex > endTokenIndex ||
+            startKyberFeeHandlerIndex > endKyberFeeHandlerIndex ||
+            supportedTokens.length == 0
+        ) {
+            // no need to do anything
+            return amounts;
+        }
+
+        uint256 endTokenId = (endTokenIndex >= supportedTokens.length) ?
+            supportedTokens.length : endTokenIndex;
+
+        for (uint256 i = startTokenIndex; i < endTokenId; i++) {
+            KyberFeeHandlerData[] memory kyberFeeHandlerArray = kyberFeeHandlersPerToken[supportedTokens[i]];
+            uint256 endKyberFeeHandlerId = (endKyberFeeHandlerIndex >= kyberFeeHandlerArray.length) ?
+                kyberFeeHandlerArray.length : endKyberFeeHandlerIndex;
+            amounts = new uint256[](endKyberFeeHandlerId - startKyberFeeHandlerIndex + 1);
```

Comment:
similar overflow?

Reply:
overflow? weeelll as long as the user inputs the right params... worst case is out of gas txns...

Reply:
guess its better to add require for start and end index.